provider/marathon.go

```diff
@@ -427,7 +426,9 @@ func (provider *Marathon) getFrontendBackend(application marathon.Application) s
 func (provider *Marathon) getSubDomain(name string) string {
 	if provider.GroupsAsSubDomains {
 		splitedName := strings.Split(strings.TrimPrefix(name, "/"), "/")
-		sort.Sort(sort.Reverse(sort.StringSlice(splitedName)))
+		for i, j := 0, len(splitedName)-1; i < j; i, j = i+1, j-1 {
+			splitedName[i], splitedName[j] = splitedName[j], splitedName[i]
+		}
```

Comment:
sort.Reverse returns the list in reverse SORTED order, not reverse order, per the unit tests added.